src/utils/mi_gpu_data.yaml

```diff
@@ -0,0 +1,144 @@
+# --------------------------------------------------------------------------------
+#
+# This yaml file tracks MI gpu data in a tree structure
+# MI GPUs
+#   |-- series: the specific MI series; mi50, mi100, mi200, mi300
+#       |-- architecture: currently, only mi300 gpus hold different architectures
+#           |-- models
+#               |-- chip_ids: chip id is specific to the environment the gpu is being used on
+#               |-- mi300_arch: mi300 specific architectures; mi300a, mi300x
+#                   |-- partition_mode: currently, only mi300 gpus hold partition mode information
+#                                       two types: compute partition mode, memory partition mode,
+#                                       currently only mi300 gpus contains compute partition mode information on number of xcds
+#
+# --------------------------------------------------------------------------------
```

Comment:
Probably, we might want to add a comment here, something like:
This file is stack holder of  some common spec for one GPU series, not all!
For example, the CU # would be based on info retrieved from other tool.